pkg/parse/namespace.go

```diff
@@ -280,8 +310,13 @@ func (p *namespace) setSyncStatusWithRetries(ctx context.Context, errs status.Mu
 		return nil
 	}
 
-	metrics.RecordReconcilerErrors(ctx, "sync", status.ToCSE(errs))
-	metrics.RecordPipelineError(ctx, configsync.RepoSyncName, "sync", rs.Status.Sync.ErrorSummary.TotalCount)
+	csErrs := status.ToCSE(errs)
+	metrics.RecordReconcilerErrors(ctx, "sync", csErrs)
+	metrics.RecordPipelineError(ctx, configsync.RepoSyncName, "sync", len(csErrs))
```

Comment:
Pipeline errors? I don't think so. That would mean they would never zero out, and the latest value would always be non-zero. They need to zero out or we can't test there are no errors after correcting the problem. They all have a different component tag value, which makes them unique. This one is component=sync.

Reply:
Sorry, I misunderstood. I agree that the reconcile errors and pipeline errors should only be sent after the sync completes, not every 5s while syncing. So I can wrap these in the !syncing flag. However, while that will stop sending every 5s, I'm not sure if it means it will only send AFTER a sync. It might also send BEFORE the sync, especially on a re-sync, when the cache is wiped out before syncing. In fact, that case might be why we have some lost errors...

Reply:
Synced offline on this. We can clean this up in a separate PR.